railties/lib/rails/application/finisher.rb

```diff
@@ -138,6 +138,9 @@ def self.complete(_state)
           # some sort of reloaders dependency support, to be added.
           require_unload_lock!
           reloader.execute
+          if app.config.eager_load
+            app.routes.eager_load!
```

Comment:
@matthewd I tried to do that in https://github.com/rails/rails/pull/27647/commits/9cf3897843c53911f467931747a2a3c4eb5388bf. Let me know what you think.